Client/game_sa/CObjectGroupPhysicalPropertiesSA.cpp

```diff
@@ -0,0 +1,372 @@
+{
+    m_pInterface = &pObjectInfo[ucObjectGroup];
+    m_bModified = MapFind(ms_OriginalGroupProperties, ucObjectGroup);    
+}
+
+CObjectGroupPhysicalPropertiesSAInterface* CObjectGroupPhysicalPropertiesSA::GetInterface()
+{
+    return m_pInterface;
+}
+
+void CObjectGroupPhysicalPropertiesSA::SetGroup(unsigned char ucObjectGroup)
+{
+    m_pInterface = &pObjectInfo[ucObjectGroup];
+    m_ucObjectGroup = ucObjectGroup;
+    m_bModified = MapFind(ms_OriginalGroupProperties, ucObjectGroup);
+}
+
+unsigned char CObjectGroupPhysicalPropertiesSA::GetGroup()
+{
+    return m_ucObjectGroup;
+}
+
+bool CObjectGroupPhysicalPropertiesSA::IsValid()
+{
+    return m_pInterface;
+}
+
+void CObjectGroupPhysicalPropertiesSA::ChangeSafeguard()
+{
+    if (m_bModified)
+        return;
+
+    m_bModified = true;
+    // Make copy of original
+    if (!MapFind(ms_OriginalGroupProperties, m_ucObjectGroup))
+    {
+        auto pOriginalCopy = new CObjectGroupPhysicalPropertiesSAInterface;
+        memcpy(pOriginalCopy, m_pInterface, sizeof(CObjectGroupPhysicalPropertiesSAInterface));
+        MapSet(ms_OriginalGroupProperties, m_ucObjectGroup, pOriginalCopy);
+    }
+}
+
+void CObjectGroupPhysicalPropertiesSA::RestoreDefault()
+{
+    if (!m_bModified)
+        return;
+
+    auto ppOriginalCopy = MapFind(ms_OriginalGroupProperties, m_ucObjectGroup);
+    dassert(ppOriginalCopy);
+    if (!ppOriginalCopy)
+        return;
+
+    auto pOriginal = *ppOriginalCopy;
+    dassert(pOriginal);
+    if (!pOriginal)
+        return;
+    
+    memcpy(m_pInterface, pOriginal, sizeof(CObjectGroupPhysicalPropertiesSAInterface));
+    m_bModified = false;
+}
+
+void CObjectGroupPhysicalPropertiesSA::RestoreDefaultValues()
+{
+    for (auto& entry : ms_OriginalGroupProperties)
+    {
+        pGame->GetObjectGroupPhysicalProperties(entry.first)->RestoreDefault();
+        delete entry.second;
+        MapRemove(ms_OriginalGroupProperties, entry.first);
+    }
+}
+
+void CObjectGroupPhysicalPropertiesSA::SetMass(float fMass)
+{
+    if (fabs(m_pInterface->fMass - fMass) < FLOAT_EPSILON)
+        return;
+
+    ChangeSafeguard();
+    m_pInterface->fMass = fMass;
+}
+
+float CObjectGroupPhysicalPropertiesSA::GetMass()
+{
+    return m_pInterface->fMass;
+}
+
+void CObjectGroupPhysicalPropertiesSA::SetTurnMass(float fTurnMass)
+{
+    if (fabs(m_pInterface->fTurnMass - fTurnMass) < FLOAT_EPSILON)
+        return;
+
+    ChangeSafeguard();
+    m_pInterface->fTurnMass = fTurnMass;
+}
+
+float CObjectGroupPhysicalPropertiesSA::GetTurnMass()
+{
+    return m_pInterface->fTurnMass;
+}
+
+void CObjectGroupPhysicalPropertiesSA::SetAirResistance(float fAirResistance)
+{
+    if (fabs(m_pInterface->fAirResistance - fAirResistance) < FLOAT_EPSILON)
+        return;
+
+    ChangeSafeguard();
+    m_pInterface->fAirResistance = fAirResistance;
+}
+
+float CObjectGroupPhysicalPropertiesSA::GetAirResistance()
+{
+    ChangeSafeguard();
```

Comment:
1. Why is `ChangeSafeguard();` called here, but not consistently across all getters?
2. Maybe also change all applicable member functions to `const`?